server/src/test/java/org/candlepin/resource/HypervisorResourceTest.java

```diff
@@ -310,4 +312,82 @@ public void checkInCreatesNoNewConsumerWhenCreateIsFalse() throws Exception {
     public void ensureBadRequestWhenNoMappingIsIncludedInRequest() {
         hypervisorResource.hypervisorUpdate(null, principal, "an-owner", false);
     }
+
+    @Test
+    public void ensureEmptyHypervisorIdsAreIgnored() throws Exception {
+        Owner owner = new Owner("admin");
+
+        Map<String, ArrayList<GuestId>> hostGuestMap = new HashMap<String, ArrayList<GuestId>>();
+        hostGuestMap.put("", new ArrayList(Arrays.asList(new GuestId("GUEST_A"),
+                                         new GuestId("GUEST_B"))));
+        hostGuestMap.put("HYPERVISOR_A", new ArrayList(Arrays.asList(new GuestId("GUEST_C"),
+                                                     new GuestId("GUEST_D"))));
+        when(ownerCurator.lookupByKey(eq(owner.getKey()))).thenReturn(owner);
+
+        when(consumerCurator.getHostConsumersMap(any(Owner.class),
+             any(Set.class))).
+             thenReturn(new VirtConsumerMap());
+        when(consumerCurator.getGuestConsumersMap(any(Owner.class), any(Set.class))).
+            thenReturn(new VirtConsumerMap());
+
+        when(ownerCurator.lookupByKey(eq(owner.getKey()))).thenReturn(owner);
+        when(principal.canAccess(eq(owner), eq(SubResource.CONSUMERS), eq(Access.CREATE))).
+            thenReturn(true);
+        when(consumerTypeCurator.lookupByLabel(
+         eq(ConsumerTypeEnum.HYPERVISOR.getLabel()))).thenReturn(hypervisorType);
+        when(idCertService.generateIdentityCert(any(Consumer.class)))
+            .thenReturn(new IdentityCertificate());
+
+        HypervisorCheckInResult result = null;
+        try {
+            result = hypervisorResource.hypervisorUpdate(hostGuestMap,
+            principal, owner.getKey(), true);
+        }
+        catch (PersistenceException e) {
+            fail("Expected no exception");
+        }
+        assertTrue(result != null);
+        assertTrue(result.getCreated().size() == 1);
+        assertTrue(!result.getCreated().toArray(new Consumer[1])[0].getHypervisorId().equals(""));
```

Comment:
Why convert to an array?  Does `result.getCreated().get(0)` not work?

And here again, use `assertEquals("", ...)` so you get a message like `Expected "" but was "Hello world"` instead of JUnit just throwing an `AssertionError` with zero context.